### PR TITLE
Rename product_type to product_namespace

### PR DESCRIPTION
**Pull request by silasdavis**
Closes https://jira.hyperledger.org/browse/GRID-97

Signed-off-by: Silas Davis <silas@monax.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# TODO: replace with root Cargo.toml?
```
Comment (vaporos) [inline Makefile]: Please exclude this file, there is work to add a top-level Cargo.toml in-progress and should appear soon-ish. (Have to work out its impact on the Dockerfiles.)

File: contracts/product/src/handler.rs [hunk 2]
```diff
@@ -121,7 +121,7 @@ impl ProductTransactionHandler {
         }
 
         // Check if the product type is a GS1 product
-        if product_type != &ProductType::GS1 {
+        if product_namespace != &ProductType::GS1 {
```
Comment (vaporos) [inline contracts/product/src/handler.rs]: Should ProductType be renamed to ProductNamespace?

Comment (Patrick-Erichsen) [reply contracts/product/src/handler.rs]: @vaporos from the Product RFC (https://github.com/hyperledger/grid-rfcs/blob/master/text/0005-product.md#product-representation)

```
message Product { 
    enum ProductNamespace { 
        UNSET_NAMESPACE = 0; 
        GS1 = 1; 
    }
    
    ProductNamespace product_namespace = 1; 
    string product_id = 2; 
    string owner = 3;
    repeated PropertyValue properties = 4; 
} 
```

Comment (adeebahmed) [reply contracts/product/src/handler.rs]: yes

File: Makefile (hunk 1 above)
Comment (adeebahmed) [reply Makefile]: 👍

**Comment by RyanLassigBanks** [comment]
Closing PR. Renaming product type to namespace has been done as a larger effort to update the product smart contract to be in line with its RFC. Please see [PR 311](https://github.com/hyperledger/grid/pull/311) for details.

**Pull request closed without merging**